pkg/storage/chunk/objectclient/client.go

```diff
@@ -22,14 +24,39 @@ var Base64Encoder = func(key string) string {
 	return base64.StdEncoding.EncodeToString([]byte(key))
 }
 
+// TenantBase64Encoder is a variation of Base64Encoder that encodes tenant
+// and remainder of the key separately and returns the results joined by "/"
+var TenantBase64Encoder = func(key string) string {
+	data := []byte(key)
+	if i := bytes.LastIndex(data, []byte("/")); i > 0 {
```

Comment:
I believe the first index instead of the last index would be a better fit as `tenant` is a prefix:
https://github.com/grafana/loki/blob/main/pkg/storage/chunk/chunk.go#L202-L215

However, both should be identical with the current chunk formats (only one `/`), but keeping in line with the `prefix` idea seems a bit more "correct" and would allow us to alter the chunk keys in the future with addtional trailing slashes if needed.

Reply:
My reasoning here was that, as you said, there are currently no `/` in the chunk-specific part of the key, but AFAIK tenant can be an arbitrary string, e.g., `division/project`. Using first index here would mean chunks of `division` would be stored together irrespective of project. Another alternative would be to just keep all `/`. IMO any of these is fine, but the behavior has to be documented.

Reply:
Great point. Let's use folders for _all_ instances of `/` in the key.